Algorithms/Mathematical-Algorithms/Power/Power_in_your_Templete.cpp

```diff
@@ -0,0 +1,32 @@
+//https://github.com/sahilbansal17/Competitive_Coding/
+
+//Name: Anil Khatri (@imkaka)
+//Email-Id: anil.soccer.khatri@gmail.com
+
+// Now we understand the Algorithm in other file we need to have it in our Templete to use Power()
+//Whenever Necessary.
+
+#include<bits/stdc++.h>
+
+using namespace std;
+
+typedef long long ll;
+ll MOD = 1e9+7;
+
+//Power Function (Recursive version) => O(long n)
+//Add this to your Templete
+//It is in one line because it has to be added in everyday used templete
+//For logic and well idented program please refer Power.cpp , Just include it in your Templete!!
+
+ll Power(ll x, ll y){ll temp;if( y == 0)return 1;temp = Power(x, y/2)%MOD;if (y%2 == 0)return (temp%MOD*temp%MOD)%MOD;else{if(y > 0)return (x%MOD*temp%MOD*temp%MOD)%MOD;else return (temp*temp)/x;}}
```

Comment:
• Explain this function properly. 
• Don't write in one line. Use appropriate spaces for mathematical expressions to be clearly understood.
• Can `(x % MOD) * (temp % MOD) * (temp % MOD)` overflow?
• `(temp * temp)/x` returned in the last else is not clear.

Reply:
Sir This is the Recursive Version of Power.cpp as and it is just a code to be added in Template, That's Why it is in one line.
* Yes ```(x % MOD) * (temp % MOD) * (temp % MOD)``` we are trying to avoid any overflow as in Competitive Programming It is generally asked for modulo by ```1e9+7```.